tests/Microsoft.Identity.Test.Common/Core/Mocks/MockHttpMessageHandler.cs

```diff
@@ -147,6 +147,10 @@ private void ValidateHeaders(HttpRequestMessage request)
             {
                 foreach (var kvp in ExpectedRequestHeaders)
                 {
+                    // Skip validation for x-ms-client-request-id since it's random
+                    if (kvp.Key.Equals("x-ms-client-request-id", StringComparison.OrdinalIgnoreCase))
```

Comment:
Is this needed? If you dont add the `x-ms-client-request-id` to the expected headers collection when creating the handler, it wont be checked here.

Reply:
resolved in latest commit